File: internal/plm/entity/sku.go
package entity

import "time"

// ProductSKU 产品SKU/配色方案
type ProductSKU struct {
	ID          string    `json:"id" gorm:"primaryKey;size:32"`
	ProjectID   string    `json:"project_id" gorm:"size:32;not null;index"`
	Name        string    `json:"name" gorm:"size:128;not null"`
	Code        string    `json:"code" gorm:"size:32"`
	Description string    `json:"description,omitempty"`
	Status      string    `json:"status" gorm:"size:16;not null;default:active"`
	SortOrder   int       `json:"sort_order" gorm:"default:0"`
	CreatedBy   string    `json:"created_by" gorm:"size:32;not null"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`

	// Relations
	Project      *Project          `json:"project,omitempty" gorm:"foreignKey:ProjectID"`
	CMFConfigs   []SKUCMFConfig    `json:"cmf_configs,omitempty" gorm:"foreignKey:SKUID"`
	BOMOverrides []SKUBOMOverride  `json:"bom_overrides,omitempty" gorm:"foreignKey:SKUID"`
}

func (ProductSKU) TableName() string {
	return "product_skus"
}

// SKUCMFConfig SKU × SBOM Item → CMF配置
type SKUCMFConfig struct {
	ID               string    `json:"id" gorm:"primaryKey;size:32"`
	SKUID            string    `json:"sku_id" gorm:"size:32;not null;index"`
	BOMItemID        string    `json:"bom_item_id" gorm:"size:32;not null"`
	Color            string    `json:"color" gorm:"size:64"`
	ColorCode        string    `json:"color_code" gorm:"size:32"`
	SurfaceTreatment string    `json:"surface_treatment" gorm:"size:128"`
	ProcessParams    string    `json:"process_params,omitempty" gorm:"type:jsonb"`
	Notes            string    `json:"notes,omitempty"`
	CreatedAt        time.Time `json:"created_at"`
	UpdatedAt        time.Time `json:"updated_at"`

	SKU     *ProductSKU     `json:"sku,omitempty" gorm:"foreignKey:SKUID"`
	BOMItem *ProjectBOMItem `json:"bom_item,omitempty" gorm:"foreignKey:BOMItemID"`
}

func (SKUCMFConfig) TableName() string {
	return "sku_cmf_configs"
}

// SKUBOMOverride SKU结构差异
type SKUBOMOverride struct {
	ID                    string    `json:"id" gorm:"primaryKey;size:32"`
	SKUID                 string    `json:"sku_id" gorm:"size:32;not null;index"`
	Action                string    `json:"action" gorm:"size:16;not null"`
	BaseItemID            *string   `json:"base_item_id,omitempty" gorm:"size:32"`
	OverrideName          string    `json:"override_name,omitempty" gorm:"size:128"`
	OverrideSpecification string    `json:"override_specification,omitempty"`
	OverrideQuantity      float64   `json:"override_quantity" gorm:"type:numeric(15,4);default:1"`
	OverrideUnit          string    `json:"override_unit,omitempty" gorm:"size:16;default:pcs"`
	OverrideMaterialType  string    `json:"override_material_type,omitempty" gorm:"size:64"`
	OverrideProcessType   string    `json:"override_process_type,omitempty" gorm:"size:32"`
	Notes                 string    `json:"notes,omitempty"`
	CreatedAt             time.Time `json:"created_at"`
	UpdatedAt             time.Time `json:"updated_at"`

	SKU      *ProductSKU     `json:"sku,omitempty" gorm:"foreignKey:SKUID"`
	BaseItem *ProjectBOMItem `json:"base_item,omitempty" gorm:"foreignKey:BaseItemID"`
}

func (SKUBOMOverride) TableName() string {
	return "sku_bom_overrides"
}

File: nimo-plm-web/src/api/sku.ts
import apiClient from './client';

// ========== Types ==========

export interface ProductSKU {
  id: string;
  project_id: string;
  name: string;
  code: string;
  description?: string;
  status: string;
  sort_order: number;
  created_by: string;
  created_at: string;
  updated_at: string;
  cmf_configs?: SKUCMFConfig[];
  bom_items?: SKUBOMItem[];
}

export interface SKUCMFConfig {
  id: string;
  sku_id: string;
  bom_item_id: string;
  color: string;
  color_code: string;
  surface_treatment: string;
  process_params?: string;
  notes?: string;
  created_at: string;
  updated_at: string;
  bom_item?: {
    id: string;
    name: string;
    item_number: number;
    specification?: string;
    material_type?: string;
    process_type?: string;
  };
}

export interface SKUBOMItem {
  id: string;
  sku_id: string;
  bom_item_id: string;
  cmf_variant_id?: string;
  quantity: number;
  notes?: string;
  created_at: string;
  updated_at: string;
  bom_item?: {
    id: string;
    name: string;
    item_number: number;
    specification?: string;
    material_type?: string;
  };
  cmf_variant?: {
    id: string;
    variant_index: number;
    material_code: string;
    color_hex: string;
    material: string;
    finish: string;
    texture: string;
    coating: string;
    pantone_code: string;
  };
}

export interface FullBOMItem {
  id: string;
  item_number?: number;
  name: string;
  specification?: string;
  quantity: number;
  unit: string;
  category?: string;
  material_type?: string;
  process_type?: string;
  is_appearance_part?: boolean;
  color?: string;
  color_code?: string;
  color_hex?: string;
  material_code?: string;
  finish?: string;
  texture?: string;
  coating?: string;
  surface_treatment?: string;
  cmf_variant_id?: string;
  cmf_variant?: {
    id: string;
    variant_index: number;
    material_code: string;
    color_hex: string;
    material: string;
    finish: string;
    texture: string;
    coating: string;
    pantone_code: string;
  };
}

// ========== API ==========

export const skuApi = {
  // SKU CRUD
  listSKUs: async (projectId: string) => {
    const res = await apiClient.get(`/projects/${projectId}/skus`);
    return res.data.data?.items || [];
  },

  createSKU: async (projectId: string, data: {
    name: string;
    code?: string;
    description?: string;
    sort_order?: number;
    bom_items?: Array<{ bom_item_id: string; cmf_variant_id?: string; quantity?: number }>;
  }) => {
    const res = await apiClient.post(`/projects/${projectId}/skus`, data);
    return res.data.data;
  },

  updateSKU: async (projectId: string, skuId: string, data: Partial<{ name: string; code: string; description: string; status: string; sort_order: number }>) => {
    const res = await apiClient.put(`/projects/${projectId}/skus/${skuId}`, data);
    return res.data.data;
  },

  deleteSKU: async (projectId: string, skuId: string) => {
    const res = await apiClient.delete(`/projects/${projectId}/skus/${skuId}`);
    return res.data;
  },

  // BOM Items (checkbox selection from PBOM)
  getBOMItems: async (projectId: string, skuId: string): Promise<SKUBOMItem[]> => {
    const res = await apiClient.get(`/projects/${projectId}/skus/${skuId}/bom-items`);
    return res.data.data?.items || [];
  },

  saveBOMItems: async (projectId: string, skuId: string, items: Array<{
    bom_item_id: string;
    cmf_variant_id?: string;
    quantity?: number;
    notes?: string;
  }>) => {
    const res = await apiClient.put(`/projects/${projectId}/skus/${skuId}/bom-items`, items);
    return res.data.data?.items || [];
  },

  // CMF Config
  getCMFConfigs: async (projectId: string, skuId: string): Promise<SKUCMFConfig[]> => {
    const res = await apiClient.get(`/projects/${projectId}/skus/${skuId}/cmf`);
    return res.data.data?.items || [];
  },

  saveCMFConfigs: async (projectId: string, skuId: string, configs: Array<{
    bom_item_id: string;
    color: string;
    color_code: string;
    surface_treatment: string;
    process_params?: string;
    notes?: string;
  }>) => {
    const res = await apiClient.put(`/projects/${projectId}/skus/${skuId}/cmf`, configs);
    return res.data.data?.items || [];
  },

  // Full BOM (merged: selected PBOM items + CMF)
  getFullBOM: async (projectId: string, skuId: string): Promise<FullBOMItem[]> => {
    const res = await apiClient.get(`/projects/${projectId}/skus/${skuId}/full-bom`);
    return res.data.data?.items || [];
  },
};
